### PR TITLE
 bwa: import from homebrew/science

### PR DESCRIPTION
**Pull request by tmozgach**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/bwa.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  desc "Burrow-Wheeler Aligner for pairwise alignment of DNA"
+  homepage "https://github.com/lh3/bwa"
+  url "https://github.com/lh3/bwa/releases/download/v0.7.16/bwa-0.7.16a.tar.bz2"
+  sha256 "8fecdb5f88871351bbe050c18d6078121456c36ad75c5c78f33a926560ffc170"
+  head "https://github.com/lh3/bwa.git"
+
+  def install
+    system "make", "CC=#{ENV.cc}", "CFLAGS=#{ENV.cflags}"
+    bin.install "bwa"
+    doc.install "README.md", "NEWS.md"
```
Comment (sjackman) [inline Formula/bwa.rb]: This line should no longer be necessary. These documentation files are installed by default now by `brew`. Can you check that after removing this line, these two files are still installed in the keg?

File: Formula/bwa.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    man1.install "bwa.1"
+  end
+
+  test do
+    (testpath/"test.fasta").write ">0\nAGATGTGCTG\n"
+    system "#{bin}/bwa", "index", "test.fasta"
+    assert_predicate testpath/"test.fasta.bwt", :exist?
+    assert_match "AGATGTGCTG", shell_output("#{bin}/bwa mem test.fasta test.fasta")
```
Comment (sjackman) [inline Formula/bwa.rb]: Thanks for adding a test, Tanya!

**Comment by tmozgach** [comment]
@sjackman Done!

**Comment by sjackman** [comment]
Is this PR good to merge?

File: Formula/bwa.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Bwa < Formula
+  desc "Burrow-Wheeler Aligner for pairwise alignment of DNA"
+  homepage "https://github.com/lh3/bwa"
+  url "https://github.com/lh3/bwa/releases/download/v0.7.16/bwa-0.7.16a.tar.bz2"
+  sha256 "8fecdb5f88871351bbe050c18d6078121456c36ad75c5c78f33a926560ffc170"
+  head "https://github.com/lh3/bwa.git"
+
+  def install
+    system "make", "CC=#{ENV.cc}", "CFLAGS=#{ENV.cflags}"
+    bin.install "bwa"
```
Comment (fxcoudert) [inline Formula/bwa.rb]: https://github.com/lh3/bwa/issues/172

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thank you @tmozgach! :white_check_mark:

**Pull request closed without merging**